nomos-core/chain-defs/src/sdp/mod.rs

```diff
@@ -249,22 +249,174 @@ impl WithdrawMessage {
     }
 }
 
-#[derive(Clone, Copy, Debug, Eq, PartialEq, Hash, Serialize, Deserialize)]
-pub struct ActiveMessage<Metadata> {
+#[derive(Clone, Debug, Eq, PartialEq, Hash, Serialize, Deserialize)]
+pub struct ActiveMessage {
     pub declaration_id: DeclarationId,
     pub nonce: Nonce,
-    pub metadata: Option<Metadata>,
+    pub metadata: Option<ActivityMetadata>,
 }
 
-impl<Metadata: AsRef<[u8]>> ActiveMessage<Metadata> {
+impl ActiveMessage {
     #[must_use]
     pub fn payload_bytes(&self) -> Bytes {
         let mut buff = BytesMut::new();
         buff.extend_from_slice(self.declaration_id.0.as_ref());
         buff.extend_from_slice(&(self.nonce.to_le_bytes()));
         if let Some(metadata) = &self.metadata {
-            buff.extend_from_slice(metadata.as_ref());
+            buff.extend_from_slice(&metadata.to_metadata_bytes());
         }
         buff.freeze()
     }
 }
+
+const METADATA_VERSION_BYTE: u8 = 0x01;
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq, Hash)]
+pub struct DaActivityProof {
+    pub current_session: SessionNumber,
+    pub previous_session_opinions: Vec<u8>,
+    pub current_session_opinions: Vec<u8>,
+}
+
+impl DaActivityProof {
+    #[must_use]
+    pub fn to_metadata_bytes(&self) -> Vec<u8> {
+        let total_size = 1 // version byte
+            + size_of::<SessionNumber>()
+            + 4 // previous_session_opinions_length (u32)
```

Comment:
We should have a type alias for the lenght we use in encodings, then use `size_of` here. This hardcoded values are hard to refactor if needed.